### PR TITLE
m

### PR DESCRIPTION
**Pull request by tivisse**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_org_chart/static/src/fields/hr_org_chart.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    /**
+     * Redirect to the sub employee form view.
+     *
+     * @private
+     * @param {MouseEvent} event
+     * @returns {Promise} action loaded
+     */
+    async _onEmployeeSubRedirect(event) {
+        const employee_id = parseInt($(event.currentTarget).data('employee-id'));
+        const type = $(event.currentTarget).data('type') || 'direct';
```
Comment (kba-odoo) [inline addons/hr_org_chart/static/src/fields/hr_org_chart.js:76]:
```suggestion
        const employee_id = parseInt(event.currentTarget.dataset.employeeId);
        const type = event.currentTarget.dataset.type || 'direct';
```

File: addons/hr_org_chart/static/src/fields/hr_org_chart.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        if (!employee_id) {
+            return {};
+        }
+        const subData = await this._getSubordinatesData(employee_id, type);
+        const domain = [['id', 'in', subData]];
+        var action = await this.orm.call('hr.employee', 'get_formview_action', [employee_id]);
+        action = _.extend(action, {
```
Comment (kba-odoo) [inline addons/hr_org_chart/static/src/fields/hr_org_chart.js:83]:
```suggestion
        action = Object.assign(action, {
```

File: addons/hr_org_chart/static/src/fields/hr_org_chart.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+        this.jsonStringify = JSON.stringify;
+
+        const recordData = this.props.record.data;
+
+        this.state = useState({'employee_id': null})
+
+        // the widget is either dispayed in the context of a hr.employee form or a res.users form
+        if (!this.state.employee_id) {
+            debugger;
```
Comment (kba-odoo) [inline addons/hr_org_chart/static/src/fields/hr_org_chart.js:115]: :eyes:

File: addons/hr_org_chart/static/src/fields/hr_org_chart.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            this.view_employee_id = null;
+        } else {
+            var orgData = await this.rpc(
+                '/hr/get_org_chart',
+                {
+                    employee_id: employeeId,
+                    context: Component.env.session.user_context,
+                }
+            );
+            if (_.isEmpty(orgData)) {
```
Comment (kba-odoo) [inline addons/hr_org_chart/static/src/fields/hr_org_chart.js:136]:
```suggestion
            if (Object.keys(orgData).length === 0) {
```

**Comment by tivisse** [comment]
Integrated into
https://github.com/odoo/odoo/pull/98301
https://github.com/odoo/enterprise/pull/30517

**Pull request closed without merging**